kai/incident_store.py

```diff
@@ -229,57 +234,195 @@ def find_solutions(
+
+    def delete_store(self):
+        self.store = {}
+
+    def load_report(self, app: Application, report: Report) -> tuple[int, int, int]:
+        """
+        Returns: (number_new_incidents, number_unsolved_incidents,
+        number_solved_incidents): tuple[int, int, int]
+        """
+        # FIXME: Only does stuff within the same application. Maybe fixed?
+
+        # create entries if not exists
+        # reference the old-new matrix
+        #           old
+        #         | NO     | YES
+        # --------|--------+-----------------------------
+        # new NO  | -      | update (SOLVED, embeddings)
+        #     YES | insert | update (line number, etc...)
+
+        repo_path = unquote(urlparse(app.repo_uri_local).path)
+        repo = Repo(repo_path)
+        old_commit: str
+        new_commit = app.current_commit
+
+        number_new_incidents = 0
+        number_unsolved_incidents = 0
+        number_solved_incidents = 0
+
+        application = self.store.setdefault(
+            app.application_name, InMemoryApplication(app.current_commit, {})
+        )
+
+        old_commit = application.current_commit
+        report_dict = report.get_report()
+
+        number_new_incidents = 0
+        number_unsolved_incidents = 0
+        number_solved_incidents = 0
+
+        for ruleset_name, ruleset_dict in report_dict.items():
+            ruleset = application.rulesets.setdefault(ruleset_name, InMemoryRuleset({}))
+
+            for violation_name, violation_dict in ruleset_dict.get(
+                "violations", {}
+            ).items():
+                violation = ruleset.violations.setdefault(
+                    violation_name, InMemoryViolation({})
+                )
+
+                store_incidents = set(violation.unsolved_incidents)
+                report_incidents = set(
+                    InMemoryIncident(
+                        x.get("uri", ""),
+                        x.get("codeSnip", ""),
+                        x.get("lineNumber", 0),
+                        x.get("variables", {}),
+                    )
+                    for x in violation_dict.get("incidents", [])
+                )
+
+                new_incidents = report_incidents.difference(store_incidents)
+                number_new_incidents += len(new_incidents)
+                for incident in new_incidents:
+                    violation.unsolved_incidents.append(incident)
+
+                unsolved_incidents = report_incidents.intersection(store_incidents)
+                number_unsolved_incidents += len(unsolved_incidents)
+
+                solved_incidents = store_incidents.difference(report_incidents)
+                number_solved_incidents += len(solved_incidents)
+                for incident in solved_incidents:
+                    file_path = os.path.join(
+                        repo_path,
+                        unquote(urlparse(incident.uri).path).removeprefix(
+                            "/tmp/source-code/"
```

Comment:
what is this directory? If it's just a placeholder I'd name it something like `/placeholder` to make it clear

Reply:
When retrieving uris from the report, some of them had "/tmp/source-code/" as their root path. Unsure where it originates from.

Reply:
ah I see, weird.